pkg/plugin/group/types/types.go

```diff
@@ -109,5 +110,10 @@ func (c Spec) InstanceHash() string {
 	hasher := sha1.New()
 	hasher.Write(stableFormat(c.Instance))
 	hasher.Write(stableFormat(c.Flavor))
-	return base64.URLEncoding.EncodeToString(hasher.Sum(nil))
+	encoded := base64.RawURLEncoding.EncodeToString(hasher.Sum(nil))
+	// Only valid characters are [a-z][0-9], _, - for support on specific platforms
+	encoded = strings.ToLower(encoded)
+	encoded = strings.Replace(encoded, "+", "-", 0)
```

Comment:
I am not so sure if it's a good idea to do these string replacements here which destroys information. 

Instead, can we just switch to Base32 encoding (https://golang.org/pkg/encoding/base32/)?  Other than the case, it should have only the safe characters: http://www.garykessler.net/library/base64.html